Packs/Cryptosim/Integrations/Cryptosim/Cryptosim.py

```diff
@@ -0,0 +1,228 @@
+    result = client.correlation_alerts()
+
+    return CommandResults(
+        outputs_prefix='CorrelationAlerts',
+        outputs_key_field='',
+        outputs=result,
+    )
+
+
+def test_module(client: Client) -> str:
+    """Tests API connectivity and authentication'
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+
+    :type client: ``Client``
+    :param Client: client to use
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    message: str = ''
+    try:
+        if client.correlations().get('StatusCode') == 200:
+            message = 'ok'
+        else:
+            raise Exception(f"""StatusCode: 
+                            {client.correlations().get('StatusCode')}, 
+                            Error: {client.correlations().get('ErrorMessage')}
+                            """)
+    except DemistoException as e:
+        if '401' in str(e):  # TODO: make sure you capture authentication errors
+            message = 'Authorization Error: make sure API User and Password is correctly set'
+        else:
+            raise e
+    return message
+
+
+''' INCIDENT '''
+
+
+def fetch_incidents(client: Client):
+
+    max_results = arg_to_number(arg=demisto.params().get('max_fetch'), arg_name='max_fetch', required=False)
+
+
+    first_fetch_time = arg_to_datetime(demisto.params().get('first_fetch')).isoformat()
+
+
+    last_run = demisto.getLastRun()
+    last_fetch = last_run.get('last_fetch', first_fetch_time)
+
+
+    incidentsList=[]
+    alert_response = client.correlation_alerts()
+    incident_data = alert_response['Data']
+
+    for inc in incident_data:
+
+
+        if len(incidentsList) > max_results:
+            break
+
+        incident_name = inc['CorrelationAlert']['NAME']
+        time_stamp = inc['CorrelationAlert']['CREATEDATE']+"Z"
+
+
+
+        severityLvl = int(inc['CorrelationAlert']['RISK'])
+        if severityLvl >=0 and severityLvl <= 5:
+            severity = 1
+        elif severityLvl > 5 and severityLvl <= 7:
+            severity = 2
+        elif severityLvl > 7 and severityLvl <= 9:
+            severity = 3
+        elif severityLvl > 9 and severityLvl <= 10:
+            severity = 4
+        else:
+            severity = 0
+
+        # "log" column is stringfyed 'Log' data.
+        inc['Log'].pop("log")
+
+        incidentObject = {**inc['Log'], **inc['CorrelationAlert']}
+
+        incident = {
+            'name': incident_name,
+            'occurred': time_stamp,
+            'rawJSON': json.dumps(incidentObject),
+            "severity": severity,
+            'type': 'Crpyotsim CorrelationAlert'
+        }
+
+        incidentsList.append(incident)
+
+
+
+    # Save the next_run as a dict with the last_fetch key to be stored
+    next_run = {'last_fetch': last_fetch}
```

Comment:
Since you are not increasing last_fetch, you will always be stuck at the same time

Reply:
```        
created_incident = datetime.strptime(time_stamp, DATE_FORMAT)
last_fetch = datetime.strptime(last_fetch,DATE_FORMAT)
if created_incident > last_fetch:
    last_fetch = created_incident.strftime(DATE_FORMAT)
```
I have creation of my alert date and compare with last_fetch. Is this well ?